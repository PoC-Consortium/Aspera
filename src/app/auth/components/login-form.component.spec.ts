import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { StoreModule, Store, combineReducers } from '@ngrx/store';
import { LoginFormComponent } from '../components/login-form.component';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';

describe('Login Page', () => {
  let fixture: ComponentFixture<LoginFormComponent>;
  let instance: LoginFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, FormsModule],
      declarations: [LoginFormComponent],
      schemas: [NO_ERRORS_SCHEMA],
    });

    fixture = TestBed.createComponent(LoginFormComponent);
    instance = fixture.componentInstance;
  });

  it('should compile', () => {
    fixture.detectChanges();

    /**
     * The login form is a presentational component, as it
     * only derives its state from inputs and communicates
     * externally through outputs. We can use snapshot
     * tests to validate the presentation state of this component
     * by changing its inputs and snapshotting the generated
     * HTML.
     *
     * We can also use this as a validation tool against changes
     * to the component's template against the currently stored
     * snapshot.
     */
    expect(fixture).toMatchSnapshot();
  });

  it('should disable the form if pending', () => {
    instance.pending = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display an error message if provided', () => {
    instance.errorMessage = 'Invalid credentials';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
