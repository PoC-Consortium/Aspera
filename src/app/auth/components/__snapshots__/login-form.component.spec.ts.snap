// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should compile 1`] = `
<bc-login-form
  submitted={[Function EventEmitter]}
>
  <div
    class="login"
  >
    <form
      class="login-inner ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      <img
        class="logo"
        src="../../../public/assets/img/burst_logo.png"
      />
      <span
        class="version"
      >
        
      </span>
      <mat-form-field
        appearance="outline"
      >
        <mat-label>
          Enter passphrase
        </mat-label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          matinput=""
          name="passphrase"
          ng-reflect-name="passphrase"
          ng-reflect-required=""
          placeholder="Enter passphrase"
          required=""
          type="password"
        />
        
        <mat-error>
          
        </mat-error>
        
      </mat-form-field>
      <button
        aria-label="Submit"
        disabled=""
        mat-button=""
      >
        Submit
      </button>
    </form>
  </div>
</bc-login-form>
`;

exports[`Login Page should disable the form if pending 1`] = `
<bc-login-form
  pending={[Function Boolean]}
  submitted={[Function EventEmitter]}
>
  <div
    class="login"
  >
    <form
      class="login-inner ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      <img
        class="logo"
        src="../../../public/assets/img/burst_logo.png"
      />
      <span
        class="version"
      >
        
      </span>
      <mat-form-field
        appearance="outline"
      >
        <mat-label>
          Enter passphrase
        </mat-label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          matinput=""
          name="passphrase"
          ng-reflect-name="passphrase"
          ng-reflect-required=""
          placeholder="Enter passphrase"
          required=""
          type="password"
        />
        
        <mat-error>
          
        </mat-error>
        
      </mat-form-field>
      <button
        aria-label="Submit"
        disabled=""
        mat-button=""
      >
        Submit
      </button>
    </form>
  </div>
</bc-login-form>
`;

exports[`Login Page should display an error message if provided 1`] = `
<bc-login-form
  errorMessage={[Function String]}
  submitted={[Function EventEmitter]}
>
  <div
    class="login"
  >
    <form
      class="login-inner ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      <img
        class="logo"
        src="../../../public/assets/img/burst_logo.png"
      />
      <span
        class="version"
      >
        
      </span>
      <mat-form-field
        appearance="outline"
      >
        <mat-label>
          Enter passphrase
        </mat-label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          matinput=""
          name="passphrase"
          ng-reflect-name="passphrase"
          ng-reflect-required=""
          placeholder="Enter passphrase"
          required=""
          type="password"
        />
        
        <mat-error>
          Invalid credentials
        </mat-error>
        
      </mat-form-field>
      <button
        aria-label="Submit"
        disabled=""
        mat-button=""
      >
        Submit
      </button>
    </form>
  </div>
</bc-login-form>
`;
